src/amo/components/Addon/index.js

```diff
@@ -173,114 +185,174 @@ export class AddonBase extends React.Component {
+      return null;
+    }
+
+    return (
+      <ShowMoreCard header={i18n.sprintf(
+        i18n.gettext('About this %(addonType)s'), { addonType }
+      )} className="AddonDescription">
+        <div className="AddonDescription-contents"
+          ref={(ref) => { this.addonDescription = ref; }}
+          {...descriptionProps}>
+          {descriptionPlaceholder}
+        </div>
+      </ShowMoreCard>
+    );
+  }
+
   render() {
     const {
       addon,
       clientApp,
+      errorHandler,
       getClientCompatibility,
       i18n,
       installStatus,
       userAgentInfo,
     } = this.props;
 
-    const authorList = addon.authors.map(
-      (author) => `<a href="${author.url}">${author.name}</a>`);
-    const description = addon.description ? addon.description : addon.summary;
-    // Themes lack a summary so we do the inverse :-/
-    // TODO: We should file an API bug about this...
-    const summary = addon.summary ? addon.summary : addon.description;
-
-    const title = i18n.sprintf(
-      // L10n: Example: The Add-On <span>by The Author</span>
-      i18n.gettext('%(addonName)s %(startSpan)sby %(authorList)s%(endSpan)s'), {
-        addonName: addon.name,
-        authorList: authorList.join(', '),
-        startSpan: '<span class="Addon-author">',
-        endSpan: '</span>',
-      });
+    let errorBanner = null;
+    if (errorHandler.hasError()) {
+      log.error('Captured API Error:', errorHandler.capturedError);
+      if (errorHandler.capturedError.responseStatusCode === 404) {
+        return <NotFound />;
+      }
+      // Show a list of errors at the top of the add-on section.
+      errorBanner = errorHandler.renderError();
+    }
 
-    const {
-      compatible, maxVersion, minVersion, reason,
-    } = getClientCompatibility({ addon, clientApp, userAgentInfo });
+    const addonType = addon ? addon.type : ADDON_TYPE_EXTENSION;
 
-    // eslint-disable react/no-danger
-    return (
-      <div className={classNames('Addon', `Addon-${addon.type}`)}>
-        <Card className="" photonStyle>
-          <header className="Addon-header">
-            <h1
-              className="Addon-title"
-              dangerouslySetInnerHTML={sanitizeHTML(title, ['a', 'span'])}
-            />
+    let authorList = [<LoadingText key="LoadingText" />];
+    if (addon) {
+      authorList = addon.authors.map(
+        (author) => `<a href="${author.url}">${author.name}</a>`);
+    }
 
-            <p
-              className="Addon-summary"
-              dangerouslySetInnerHTML={sanitizeHTML(summary, ['a'])}
-            />
+    const summaryProps = {};
+    let summaryPlaceholder = null;
+    if (addon) {
+      // Themes lack a summary so we do the inverse :-/
+      // TODO: We should file an API bug about this...
+      const summary = addon.summary ? addon.summary : addon.description;
+      summaryProps.dangerouslySetInnerHTML = sanitizeHTML(summary, ['a']);
+    } else {
+      summaryPlaceholder = <LoadingText />;
```

Comment:
If someone thinks they're clever and tries to remove this else block like I did locally, and just set the placeholder, we'll get an invariant error because React won't allow `.children` and `.dangerousSetInnerHTML`. I think it might be nice to just remove the placeholder variable and set `summaryProps.children = <LoadingText />` here.